tools/amd_build/build_amd.py

```diff
@@ -78,6 +78,7 @@
     "aten/src/ATen/cuda/*",
     "aten/src/ATen/native/cuda/*",
     "aten/src/ATen/native/cudnn/*",
+    "aten/src/ATen/native/miopen/*",
```

Comment:
Why was this needed?

Reply:
Internally, I added this to convert `#include <ATen/cuda/CUDAConfig.h>` into `#include <ATen/hip/HIPConfig.h>`, but if OSS does not require this (and obviously it is not needed), I have to understand the OSS build flow more. Please let me investigate it. Thank you for pointing this out, @jeffdaily !

Reply:
I investigated the background a bit more. This is the current situation:
- "aten/src/ATen/native/miopen/*" is now not hipified (because this is not in `build_amd.py` above)
- `#include <ATen/cuda/CUDAConfig.h>` is directly exposed for HIP compilation (e.g., in `BatchNorm_miopen.cpp`)
-  Currently `ATen/cuda/CUDAConfig.h` = `ATen/hip/HIPConfig.h` and both are there , so it works without any problem for OSS build.

(Internally, we remove `cuda/*` for HIP compilation, which causes this issue here.)

---

As far as I built ATen for ROCm, this is the only code that includes unhipified `cuda/CUDAConfig.h` for HIP compilation. @jeffdaily Should I keep this change in this PR? Because compilation succeeds now, should I keep this part untouched?

(Note: changes in `gen.py` etc, which generate `RegisterCUDA.cpp`, are to remove an unhipified `ATen/cuda/EmptyTensor.h` path from HIP compilation, but `RegisterCUDA.cpp` is build successfully without such a change since now `ATen/cuda/EmptyTensor.h` can be compiled with HIP.)

Reply:
Adding `aten/src/ATen/native/miopen/*` to the hipify script seems odd since it should be a directory containing fully-custom HIP code under source control.  Does it make sense to use `#include <ATen/hip/HIPConfig.h>` directly in any files under that `miopen/*` directory instead, rather than relying on hipify to convert the name of the CUDA header that really isn't expected to be there in the first place?

Reply:
Thanks @jeffdaily.  I truly agree that this design looks nonsense, but it is nontrivial to use `#include <ATen/hip/HIPConfig.h>` instead of `#include <ATen/cuda/CUDAConfig.h>`. This `#include <ATen/cuda/CUDAConfig.h>` is also used for CUDA to know if MIOpen is enabled. I would like to know your idea.

---

Details: To create `miopen` function stubs for CUDA, `native/miopen/*` files (e.g., `BatchNorm_miopen.cpp`) use `AT_ROCM_ENABLED()`, which is defined in `cuda/CUDAConfig`.
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/miopen/BatchNorm_miopen.cpp#L9
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/CUDAConfig.h.in#L11

This is the same design pattern as `cudnn`, for example. In this case, it looks natural since it includes `cuda/CUDAConfig.h` to get `AT_CUDNN_ENABLED()`.
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/BatchNorm.cpp#L6

Reply:
Thank you for the details.  In that case, your PR changes should be okay.  Can you rebase and make sure the ROCm CI is passing?  Your PR picked up a bad commit that has since been backed out to fix.

Reply:
When the file `aten/src/ATen/native/miopen/BatchNorm_miopen.cpp` is hipified, it will end up one directory deeper at `aten/src/ATen/native/miopen/hip/BatchNorm_miopen.cpp`.  Which file is compiled by CMake?  When I build your PR locally I get the following object file, which indicates to me that your hipified version of BatchNorm_miopen.cpp is not the source file that gets compiled.

`./build/caffe2/CMakeFiles/torch_hip.dir/__/aten/src/ATen/native/miopen/BatchNorm_miopen.cpp.o`

Reply:
Thanks. Please let me update this PR to address the issue (and I'll make sure the ROCm CI passes).

Reply:
I updated `CMakeLists.txt`. Now it correctly compiles `miopen/hip/*`.

From the ROCm CI:
```
21:06:12 [ 88%] Building CXX object caffe2/CMakeFiles/torch_hip.dir/__/aten/src/ATen/native/miopen/hip/BatchNorm_miopen.cpp.o
21:06:12 [ 88%] Building CXX object caffe2/CMakeFiles/torch_hip.dir/__/aten/src/ATen/native/miopen/hip/RNN_miopen.cpp.o
```

Reply:
@jeffdaily I'd appreciate it if you could review the change.

Reply:
Please revert.

Reply:
Thanks. Reverted the changes.